cmd/posix-errors.go

```diff
@@ -91,8 +85,7 @@ func isSysErrNotDir(err error) bool {
 // Check if the given error corresponds to the ENAMETOOLONG (name too long).
 func isSysErrTooLong(err error) bool {
 	if pathErr, ok := err.(*os.PathError); ok {
-		switch pathErr.Err {
-		case syscall.ENAMETOOLONG:
+		if pathErr.Err == syscall.ENAMETOOLONG {
```

Comment:
Further simplification is possible simply do `return pathErr.Err == syscall.ENAMETOOLONG`